File: Exercitar python/Exercicio07.py
#Função dada: f(x) = x^3 - x^2 - 13x + 8
def f(x):
    return ( (x*x*x)-(x*x)-(13*x)+8 )

print("Insira o limite inferior e superior:")
a = float(input()) #limite inferior
b = float(input()) #limite superior
c = ((a+b)/2)      #ponto médio

#Considerações iniciais
if( f(a)*f(b) >= 0):
    print("Os valores não tem sínais opostos.")
    print("Logo, não se pode garantir a existência de raízes.")
    quit()

#Looop do calculo da resposta
num_iteracoes = 0
while( f(c) != 0 and f(c) > 0.00000001 and num_iteracoes != 5):
    num_iteracoes = num_iteracoes + 1 
    #Caso: f(c) e f(a) tem sinais opostos -> a raiz está entre a e c
    if (f(c)*f(a) < 0):
        b = c 
    #Consequencia: f(c) e f(b) tem sínais opostos -> a raiz está entre c e b
    else: 
        a = c
    c = ((a+b)/2)

#Saída do programa
print("A raíz é {:.2f}, calculada com {:.2f} iterações"
        .format(f(c),num_iteracoes))

File: Exercitar java/Exercicio07/Exercicio07.java
import java.lang.Math;
class Main{
    //Função dada 
    static double f(double x){
        double x2 = x*x, x3 = x*x*x;    
        return (x3 - x2 - (13*x) + 8);
    }


    public static void main(String[] args) throws Exception {

        //Leitura dos valores do dominío 
        double a = 0, b = 0;
        System.out.println("ínsira o primeiro valor do dominio: ");
        a = EntradaTeclado.leDouble();
        System.out.println("ínsira o segundo valor do dominio: ");
        b = EntradaTeclado.leDouble();

        System.out.printf("f(a) = %f\n",f(a));
        System.out.printf("f(b) = %f\n",f(b));

        //Verificar se os números iniciais tem sinais opostos
        if( f(a)*f(b) >= 0){
            System.out.println("Os valores não tem sínais opostos.");
            System.out.println("Logo, não se pode garantir a existência de raízes.");
            return;
        }

        //loop de resolução
        double pontoMedio = (a+b)/2, extremoAux = 0;

        while( Math.abs(pontoMedio - extremoAux) > 0.0000001){
            //Caso: ponto médio e a tenham sinais opostos
            if( f(pontoMedio)*f(a) < 0){
                extremoAux = a;
                b = pontoMedio;
                pontoMedio = (a+pontoMedio)/2;
            }
            //Consequencia: ponto médio e b tem sínais opostos
            else{
                extremoAux = b;
                a = pontoMedio;
                pontoMedio = (b+pontoMedio)/2;
            }
        }
        System.out.printf("A raiz encontrada é: %f\n",pontoMedio);
    }
}